src/calc_ion_temperature.cpp

```diff
@@ -1,22 +1,123 @@
+  int64_t nAlts = grid.get_nAlts();
+
+  arma_vec temp1d(nAlts);
+  arma_vec lambda1d(nAlts);
+  arma_vec front1d(nAlts);
+  arma_vec dalt1d(nAlts);
+  arma_vec conduction1d(nAlts);
+
+  arma_cube tempT(nLons, nLats, nAlts);
+  arma_cube tempD(nLons, nLats, nAlts);
+
+  // Get the time step size
+  precision_t dt = time.get_dt();
+  
+  // Loop over all species or assume only bulk calculation
+  if (input.get_do_calc_bulk_ion_temp()==true)  nSpecs=1; // First ion specie only, is O+?
+  if (input.get_do_calc_bulk_ion_temp()==false) nSpecs=nIons;
+  
+  if (report.test_verbose(4)){
+    std::cout << "Bulk ion temp flag: " << input.get_do_calc_bulk_ion_temp() 
+              << " so 'number of ions' is " << nSpecs << "\n";
+  } 
+
+  // Loop over all species or assume only bulk calculation
+  for (iIon = 0; iIon < nSpecs; iIon++) {
+
+    for (iLon = 0; iLon < nLons; iLon++) {
+      for (iLat = 0; iLat < nLats; iLat++) {
+
+        // --------------------------------------------------------------------------
+        // Calculate heat flux (conduction) in 1D; loop over all lat,lon positions
+        // --------------------------------------------------------------------------
+        temp1d   = species[iIon].temperature_scgc.tube(iLon, iLat);         // ion temperature
+	lambda1d = 25.0 * pow(cKB,2) * pow(temp1d,2.5) / species[iIon].mass 
+		      / species[iIon].nu_ion_ion[iIon] / 8.0;               // thermal conductivity
+	front1d  = 2.0/species[iIon].density_scgc.tube(iLon,iLat)/cKB/3.0;  // front matter of term 
+	dalt1d   = grid.dalt_lower_scgc.tube(iLon, iLat);                   // grid thing for solver
+       
+	conduction1d.zeros();    // set temp variable to zero
+
+        conduction1d = solver_conduction(temp1d, lambda1d, front1d, dt, dalt1d);
+        
+	// The conduction solver gives Tnew-Told, so divide by dt
+        conduction_scgc.tube(iLon, iLat) = conduction1d / dt;
+
+      } // Lats
+    } // Lons
+
+    // --------------------------------------------------------------------------
+    // Add temperature terms together to advance ion temperature
+    // As more temperature terms get coded, they are added to the parenthesis 
+    // for inclusion in the advancement of the ion temperature
+    // --------------------------------------------------------------------------
+    if (input.get_do_calc_bulk_ion_temp()==false) {
+      species[iIon].temperature_scgc = species[iIon].temperature_scgc + 
+                                         dt * (conduction_scgc);
+    }
+  } //ions
+
+  if (input.get_do_calc_bulk_ion_temp()==false) {
+    // Use the density averaged temperature to fill the bulk temperature 
+    tempT.zeros();
+    tempD.zeros();
+    for (iIon = 0; iIon < nIons; iIon++) {
+      tempT = tempT + (species[iIon].temperature_scgc % species[iIon].density_scgc);
+      tempD = tempD + species[iIon].density_scgc;
+    }
+    temperature_scgc = tempT / tempD;
+  }
+
+  if (input.get_do_calc_bulk_ion_temp()==true) {
```

Comment:
Your spacing for operator and evaluations is inconsistent.  I think it looks nicer to have whitespace, as used in the suggestion, but either way recommend being consistent everywhere.
```suggestion
  if (input.get_do_calc_bulk_ion_temp() == true) {
```

Reply:
Need to run the astyle code on this.  It will fix all of these.

Reply:
Ran astyle, it is better now.